src/main/scala/com/johnsnowlabs/nlp/training/CoNLL.scala

```diff
@@ -290,14 +305,18 @@ case class CoNLL(
   }
 
   def schema: StructType = {
+    val docId = StructField(conllDocIdCol, StringType)
     val text = StructField(conllTextCol, StringType)
     val doc = getAnnotationType(documentCol, AnnotatorType.DOCUMENT)
     val sentence = getAnnotationType(sentenceCol, AnnotatorType.DOCUMENT)
     val token = getAnnotationType(tokenCol, AnnotatorType.TOKEN)
     val pos = getAnnotationType(posCol, AnnotatorType.POS)
     val label = getAnnotationType(labelCol, AnnotatorType.NAMED_ENTITY)
 
-    StructType(Seq(text, doc, sentence, token, pos, label))
+    if (includeDocId)
```

Comment:
I would keep the schema always the same, consider consumers of this data. Changing output schema according to input data probably is not a good idea. If docId is not available, then make them all docId==0. What do you think?

Reply:
That was the original idea in the very first commit. But then the optionality was required.